Source/Fuse.Navigation/Router.ScriptClass.uno

```diff
@@ -175,21 +175,21 @@ namespace Fuse.Navigation
 				- `relative`: An optional node that indicates the path is relative to this router outlet. The path is specified like in `gotoRelative`
 				- `style`: The style of the operation, which can be used as a matching criteria in transitions.
 		*/
-		static void Modify(Context c, Router r, object[] args)
+		static object Modify(Context c, Router r, object[] args)
 		{
-			if (!r.IsRootingCompleted) return;
+			if (!r.IsRootingCompleted) return null;
 			
 			if (args.Length != 1)
 			{
 				Fuse.Diagnostics.UserError( "`Router.modify` takes one argument", r );
-				return;
+				return null;
 			}
 			
-			var obj = args[0] as Fuse.Scripting.Object;
+			var obj = args[0] as IObject;
 			if (obj == null)
 			{
 				Fuse.Diagnostics.UserError( "`Router.modify` should be passed an object", r );
-				return;
+				return null;
 			}
 			
 			var request = new ScriptRouterRequest(c);
```

Comment:
I'm assuming this function is currently running in the JS thread, meaning it's arguments haven't be reflected into the UI thread yet. I don't see any changes in the `RouterRequest` that does the reflection. Am I missing something?

Reply:
Yeah, this method might have been incorrectly ported.

Reply:
Yeah, I think I have a better version now. I'll fix up some other small things an push out...

Reply:
Done.